core/trino-main/src/main/java/io/trino/metadata/FunctionRegistry.java

```diff
@@ -601,7 +601,8 @@ public FunctionRegistry(
                 .functions(MAX_AGGREGATION, MIN_AGGREGATION, new MaxNAggregationFunction(blockTypeOperators), new MinNAggregationFunction(blockTypeOperators))
                 .function(COUNT_COLUMN)
                 .functions(JSON_TO_ROW, JSON_STRING_TO_ROW, ROW_TO_ROW_CAST)
-                .functions(VARCHAR_CONCAT, VARBINARY_CONCAT)
+                .functions(VARCHAR_CONCAT_FUNCTIONS)
```

Comment:
full exception message is rather long
```
$ bin/trino-cli --execute 'SELECT concat('13', 10)' 2>&1 | wc
     47   98078 1400863
```

-- it's going to be problematic for end users.
why would we be doing that?

Reply:
@findepi The main reason is following ANSI SQL specification about string concatenation oepration. The return type of `'a' || 'b'` should contain length instead of `varchar`. Also, this change fixes query failure when it has char type:
```
trino> SELECT 'a' || 'b' || CAST('c' as CHAR(1));
Query ... failed: line 1:19: CHAR length must be in range [0, 65536], got 65537
SELECT 'a' || 'b' || CAST('c' as CHAR(1))
```

Reply:
@ebyhr 

`'a'` is of type `varchar(1)`

`varchar(1) || varchar(1) -> varchar` -- this is OK and yes, we could do better, i.e. know the type is `varchar(2)`. We should be able to do this in some more reasonable manner than loading a ton of overloads of a function into the system. If we need better type resolution for function results, this is what we should do. 

`varchar || char(1) -> char(65537)` -- this fails (char limit overflow). Your PR doesn't fix that, does it?
This should produce `varchar` instead.

Reply:
> Your PR doesn't fix that, does it?

@findepi Right, this PR doesn't fix it. Actually, I was trying to fix the both situation, but I sent this PR for the left operator. Let me look into the right operator.